ocean_lib/data_provider/data_service_provider.py

```diff
@@ -103,7 +102,100 @@ def encrypt(
 
     @staticmethod
     @enforce_types
-    def sign_message(wallet: Wallet, msg: str) -> str:
+    def initialize(
+        did: str,
+        service_id: str,
+        file_index: int,
+        consumer_address: str,
+        service_endpoint: str,
+        userdata: Optional[Dict] = None,
+    ) -> Response:
+        initialize_endpoint = service_endpoint
```

Comment:
please use either preparedUrl or pass the GET query as a parameter. We are not doing '90s PHP here, building URLs manually.